src/pool/pool_jemalloc.c

```diff
@@ -382,62 +425,91 @@ static void *op_aligned_alloc(void *pool, size_t size, size_t alignment) {
 
 static umf_result_t op_initialize(umf_memory_provider_handle_t provider,
                                   void *params, void **out_pool) {
-    (void)params; // unused
     assert(provider);
     assert(out_pool);
 
     extent_hooks_t *pHooks = &arena_extent_hooks;
     size_t unsigned_size = sizeof(unsigned);
     int err;
+    umf_jemalloc_pool_params_t *jemalloc_params =
+        (umf_jemalloc_pool_params_t *)params;
 
-    jemalloc_memory_pool_t *pool =
-        umf_ba_global_alloc(sizeof(jemalloc_memory_pool_t));
-    if (!pool) {
-        return UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+    size_t n_arenas = 0;
+    if (jemalloc_params) {
+        n_arenas = jemalloc_params->n_arenas;
     }
 
-    pool->provider = provider;
-
-    unsigned arena_index;
-    err = je_mallctl("arenas.create", (void *)&arena_index, &unsigned_size,
-                     NULL, 0);
-    if (err) {
-        LOG_ERR("Could not create arena.");
-        goto err_free_pool;
+    if (n_arenas == 0) {
+        n_arenas = utils_get_num_cores() * 4;
     }
-
-    // setup extent_hooks for newly created arena
-    char cmd[64];
-    snprintf(cmd, sizeof(cmd), "arena.%u.extent_hooks", arena_index);
-    err = je_mallctl(cmd, NULL, NULL, (void *)&pHooks, sizeof(void *));
-    if (err) {
-        snprintf(cmd, sizeof(cmd), "arena.%u.destroy", arena_index);
-        (void)je_mallctl(cmd, NULL, 0, NULL, 0);
-        LOG_ERR("Could not setup extent_hooks for newly created arena.");
-        goto err_free_pool;
+    if (n_arenas > MALLOCX_ARENA_MAX) {
```

Comment:
couldn't this be checked already in `umfJemallocPoolParamsSetNumArenas`? to get early info

Also, this here could be then updated to trim down to max instead of getting error...?

Reply:
In other pools/providers, we report configuration errors during the initialization phase, as this is the only time when full validation can be done (e.g., checking for inconsistencies between parameters).

Reply:
This is consistent with other pools and providers. I agree that we could check it elier but this is subiejct for other refactorings. 

This check is to cover the case that we could have integer overflow in allocation next line. Anyway we do not know how much arenas more we can create, as this is a global jemalloc limit, so we have to try to create them and check if it's successful